src/right_click_type.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file right_click_type.h Types related to right click options. */
+#ifndef RIGHT_CLICK_TYPE_H
+#define RIGHT_CLICK_TYPE_H
+
+#include "core/enum_type.hpp"
+
+enum RightClickClose : uint8 {
```

Comment:
This enum should either include a prefix `RCC_` on each value or be declared as `enum class` and use Pascal case values.